src/coreclr/src/gc/gc.cpp

```diff
@@ -2053,28 +2053,35 @@ ptrdiff_t round_down (ptrdiff_t add, int pitch)
 #error if GROWABLE_SEG_MAPPING_TABLE is defined, SEG_MAPPING_TABLE must be defined
 #endif
 
+// Returns true if moving an object from old_loc to new_loc respects large (double) alignment.
+// old_loc and new_loc are expected to be aligned by DATA_ALIGNMENT.
 inline
-BOOL same_large_alignment_p (uint8_t* p1, uint8_t* p2)
+BOOL same_large_alignment_p (uint8_t* old_loc, uint8_t* new_loc)
 {
 #ifdef RESPECT_LARGE_ALIGNMENT
-    return ((((size_t)p1 ^ (size_t)p2) & 7) == 0);
-#else
-    UNREFERENCED_PARAMETER(p1);
-    UNREFERENCED_PARAMETER(p2);
+    return ((((size_t)old_loc ^ (size_t)new_loc) & DATA_ALIGNMENT) == 0) || (old_loc == 0);
+#else // RESPECT_LARGE_ALIGNMENT
+    UNREFERENCED_PARAMETER(old_loc);
+    UNREFERENCED_PARAMETER(new_loc);
     return TRUE;
-#endif //RESPECT_LARGE_ALIGNMENT
+#endif // RESPECT_LARGE_ALIGNMENT
 }
 
+// Returns the padding size required to respect large (double) alignment.
```

Comment:
This is a bit confusing. I think this applies not only to doubles. It is more general requirement to align 64bit values on 8 byte locations on 32bit system when on ARM32

Reply:
@VSadov This feature is not about alignment of System.Double objects, it is about of **double alignment** of objects.  It was introduced in 2003, many years before any ARM32 work.

Reply:
I have never seen this used for anything else. 

The description also says
```
// We need the following feature as part of keeping 64-bit types aligned in the GC heap.
#define RESPECT_LARGE_ALIGNMENT //used to keep "double" objects aligned during
                                //relocation
```
It is more clear now why these comments refer to "double". I always thought the comments use `System.Double` as an example.

Reply:
@VSadov I agree this historical comment is confusing now and worth updating.

Reply:
Fixed the comments:
```C++
#define RESPECT_LARGE_ALIGNMENT //Preserve double alignment of objects during relocation
```
and
```C++
// Returns true if two pointers have the same large (double than normal) alignment.
```